src/lib/types.ts

```diff
@@ -0,0 +1,11 @@
+import {
+  Locale,
+  DrupalMenuLinkContent,
+} from "next-drupal"
+
+export declare type HeaderProps = {
```

Comment:
Is there are reason you are using type instead of Interface. I usually use Interfaces for data shapes like object.